js/ui/switch.js

```diff
@@ -281,31 +288,18 @@ var Switch = Editor.inherit({
         this._renderInkWave(this._$handle, e, value, 1);
     },
 
-    _getMarginBound: function() {
-        if(!this._marginBound) {
-            this._marginBound = this._$switchContainer.outerWidth(true) - this._getHandleWidth();
-        }
-        return this._marginBound;
-    },
-
-    _marginDirection: function() {
-        return this.option("rtlEnabled") ? "Right" : "Left";
-    },
-
     _offsetDirection: function() {
         return this.option("rtlEnabled") ? -1 : 1;
     },
 
     _renderPosition: function(state, swipeOffset) {
-        if(!windowUtils.hasWindow()) {
-            return;
-        }
-
-        var marginDirection = this._marginDirection(),
-            resetMarginDirection = marginDirection === "Left" ? "Right" : "Left";
+        var innerOffset = this._getInnerOffset(state, swipeOffset),
+            handleOffset = this._getHandleOffset(state, swipeOffset);
 
-        this._$switchInner.css("margin" + marginDirection, this._getCalcOffset(state, swipeOffset));
-        this._$switchInner.css("margin" + resetMarginDirection, 0);
+        if(this.option("_animateHandle")) {
+            this._$switchInner.css("transform", " translateX(" + innerOffset + ")");
```

Comment:
Maybe better to use `translator` utility here?

Reply:
'translator.move' need target position. I would like to avoid any position calculations in Switch. On the other hand 'css.transform' operates with a percentage value.